### PR TITLE
proxy for K8s

### PR DESCRIPTION
**Pull request by areed**

### CODE REVIEW

File: install_scripts/templates/kubernetes-init.sh [hunk 1]
```diff
@@ -384,6 +392,19 @@ if [ -z "$PRIVATE_ADDRESS" ]; then
...
+    fi
+
+    if [ -z "$PROXY_ADDRESS" ]; then
+        promptForProxy
+    fi
+fi
+
+exportProxy
+# kubeadm requires this in the environment to reach the K8s API server
+export no_proxy="$PRIVATE_ADDRESS,$SERVICE_CIDR" # default service cidr
```
Comment (dexhorthy) [inline install_scripts/templates/kubernetes-init.sh]: I didn't think no_proxy supported CIDR blocks. I guess it probably depends on the implementation.

**Comment by areed** [comment]
Go's net/http package doesn't but kubeadm uses the kubernetes implementation which does https://github.com/kubernetes/kubernetes/blob/07e849c98698b19de636d6f7282dbc0b2edebde7/staging/src/k8s.io/apimachinery/pkg/util/net/http.go#L86-L88

**Comment by dexhorthy** [comment]
oh wow. nifty!

**Comment by dexhorthy** [comment]
makes sense to me. I dont see where we set things up for pulling from the onprem registry -- guessing its already baked into our existing no_proxy sutff

**Pull request merged**